crates/sui-types/src/zk_login_authenticator.rs

```diff
@@ -108,13 +111,18 @@ impl AuthenticatorTrait for ZkLoginAuthenticator {
         T: Serialize,
     {
         if aux_verify_data.verify_legacy_zklogin_address {
-            if author != self.try_into()? && author != SuiAddress::legacy_try_from(self)? {
+            // If the verify_legacy_zklogin_address flag is set, then try to derive with both ways (padded and unpadded address_seed) as valid signature.
+            if author != SuiAddress::try_from_padded(&self.inputs)?
+                && author != SuiAddress::try_from_unpadded(&self.inputs)?
+            {
                 return Err(SuiError::InvalidAddress);
             }
-        } else if author != self.try_into()? {
+        } else if author != SuiAddress::try_from_unpadded(&self.inputs)? {
```

Comment:
shouldn't this be `try_from_padded`?

Reply:
sorry this is not clear, legacy == padded. if the flag is true, verify for both. if the flag is false, only verify as unpadded, which is what self.try_into() was derived to do